tests/jest/api/RestApi.spec.js

```diff
@@ -1,19 +1,675 @@
+  response = await client.fetch('/requests/study');
+  expect(response.statusCode).toBe(HttpStatus.OK.statusCode);
+  fetchedStudyRequests = response.result;
+  expect(fetchedStudyRequests).toEqual([persistedStudyRequest]);
+
+  // supervisors can fetch all
+  client.setUser(supervisor);
+  response = await client.fetch('/requests/study');
+  expect(response.statusCode).toBe(HttpStatus.OK.statusCode);
+  fetchedStudyRequests = response.result;
+  expect(fetchedStudyRequests).toEqual([persistedStudyRequest]);
+
+  // update study request fields
+  persistedStudyRequest.reasons = [StudyRequestReason.TSC];
+  persistedStudyRequest.serviceRequestId = '12345';
+
+  // cannot update non-existent study request
+  client.setUser(requester);
+  response = await client.fetch('/requests/study/1234567890', {
+    method: 'PUT',
+    data: persistedStudyRequest,
+  });
+  expect(response.statusCode).toBe(HttpStatus.NOT_FOUND.statusCode);
+
+  // other ETT1s cannot update this request
+  client.setUser(ett1);
+  response = await client.fetch(`/requests/study/${persistedStudyRequest.id}`, {
+    method: 'PUT',
+    data: persistedStudyRequest,
+  });
+  expect(response.statusCode).toBe(HttpStatus.FORBIDDEN.statusCode);
+
+  // cannot update a transient instance
+  client.setUser(requester);
+  response = await client.fetch(`/requests/study/${persistedStudyRequest.id}`, {
+    method: 'PUT',
+    data: transientStudyRequest,
+  });
+  expect(response.statusCode).toBe(HttpStatus.BAD_REQUEST.statusCode);
+
+  // requester can update study request
+  response = await client.fetch(`/requests/study/${persistedStudyRequest.id}`, {
+    method: 'PUT',
+    data: persistedStudyRequest,
+  });
+  expect(response.statusCode).toBe(HttpStatus.OK.statusCode);
+  expect(response.result.studyRequest.id).toEqual(persistedStudyRequest.id);
+  expect(response.result.studyRequestChange).toBeNull();
+  persistedStudyRequest = response.result.studyRequest;
+
+  response = await client.fetch(`/requests/study/${persistedStudyRequest.id}`);
+  expect(response.statusCode).toBe(HttpStatus.OK.statusCode);
+  fetchedStudyRequest = response.result;
+  expect(fetchedStudyRequest).toEqual(persistedStudyRequest);
+  expect(fetchedStudyRequest.lastEditorId).toEqual(requester.id);
+
+  // update more study request fields and set urgent
+  persistedStudyRequest.daysOfWeek = [3, 4];
+  persistedStudyRequest.hours = StudyHours.SCHOOL;
+  persistedStudyRequest.notes = 'oops, this is actually a school count';
+  persistedStudyRequest.urgent = true;
+  persistedStudyRequest.urgentReason = 'because I said so';
+
+  // supervisor can update study request
+  client.setUser(supervisor);
+  response = await client.fetch(`/requests/study/${persistedStudyRequest.id}`, {
+    method: 'PUT',
+    data: persistedStudyRequest,
+  });
+  expect(response.statusCode).toBe(HttpStatus.OK.statusCode);
+  expect(response.result.studyRequestChange).toBeNull();
+  persistedStudyRequest = response.result.studyRequest;
+
```

Comment:
we're actually missing a _lot_ of the branches in `PUT /requests/study/:id`, but those can be added later.